File: 卡码/代码随想录/Java/src/com/kurumi/codeThoughts/动态规划/_05_子序列问题/_01_不连续子序列/LeetCode_1035.java
package com.kurumi.codeThoughts.动态规划._05_子序列问题._01_不连续子序列;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 1035. 不相交的线
 */
public class LeetCode_1035 {
    static class Solution {
        public int maxUncrossedLines(int[] nums1, int[] nums2) {
            int[][] dp = new int[nums1.length + 1][nums2.length + 1];
            for (int i = 1; i < nums1.length + 1; i++)
                for (int j = 1; j < nums2.length + 1; j++)
                    if (nums1[i - 1] == nums2[j - 1])
                        dp[i][j] = dp[i - 1][j - 1] + 1;
                    else
                        dp[i][j] = Math.max(dp[i][j - 1], dp[i - 1][j]);
            return dp[nums1.length][nums2.length];
        }
    }
}

File: 卡码/代码随想录/1/09-动态规划/Js/05-子序列问题/01-子序列(不连续)/LeetCode_1035.js
//1035. 不相交的线
const maxUncrossedLines = function(nums1, nums2) {
	const dp=new Array(nums1.length+1).fill(0).map(item=>new Array(nums2.length+1).fill(0));
	for(let i=1;i<nums1.length+1;i++)
		for(let j=1;j<nums2.length+1;j++)
			if(nums1[i-1]==nums2[j-1]) dp[i][j]=dp[i-1][j-1]+1;
			else dp[i][j]=Math.max(dp[i-1][j],dp[i][j-1]);
	return dp[nums1.length][nums2.length];
};
